### ISSUE TITLE
DBAPI cursor receiving 2 arguments instead of 1

### ISSUE THREAD

**Issue by Jamiel-Carter** [Mause/duckdb_engine#493]
### What happened?

SQLAlchemy==1.4.45

Called `connection.execute` with two arguments: `statement=TextClause`, `parameters=[{...}]`. This resulted in the StatementError described in the `Relevant log output`. The relevant code where the error occurs is below:

```python
# base.py/_ConnectionFairy/cursor line 1111
def cursor(self, *args, **kwargs):
        """Return a new DBAPI cursor for the underlying connection.

        This method is a proxy for the ``connection.cursor()`` DBAPI
        method.

        """
        return self.dbapi_connection.cursor(*args, **kwargs)
```
When I inspect args, I see what looks like a memory address: something like `('c_7fbb330ce780_1',)`

### DuckDB Engine Version

duckdb-engine==0.6.5

### DuckDB Version

duckdb==0.6.1

### Relevant log output

```shell
Exception has occurred: StatementError
(builtins.TypeError) cursor() takes 1 positional argument but 2 were given
[SQL: INSERT INTO schema.table
(id, created, updated, ...)
VALUES (?, ?, ?, ...)]
[parameters: [{'id': '1', 'created': '2006-07-24T12:12:46Z', 'updated': '2022-09-23T15:54:17Z', ...}]]
File "/workspaces/idw/target-duckdb/.venv/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1800, in _execute_context
    context = constructor(
  File "/workspaces/idw/target-duckdb/.venv/lib/python3.8/site-packages/sqlalchemy/engine/default.py", line 1020, in _init_compiled
    self.cursor = self.create_cursor()
  File "/workspaces/idw/target-duckdb/.venv/lib/python3.8/site-packages/sqlalchemy/engine/default.py", line 1388, in create_cursor
    return self.create_server_side_cursor()
  File "/workspaces/idw/target-duckdb/.venv/lib/python3.8/site-packages/sqlalchemy/dialects/postgresql/psycopg2.py", line 636, in create_server_side_cursor
    return self._dbapi_connection.cursor(ident)
  File "/workspaces/idw/target-duckdb/.venv/lib/python3.8/site-packages/sqlalchemy/pool/base.py", line 1111, in cursor
    return self.dbapi_connection.cursor(*args, **kwargs)
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Mause** [comment]
I'm not quite sure how you've managed to trigger it, but it looks like it's breaking on trying to use server side cursors (which DuckDB doesn't support). I've put up a PR to disable them for now, but some sample code that triggers your issue would be appreciated 😁

**Comment by Jamiel-Carter** [comment]
@Mause Thank you for the speedy response. It looks like this one is user error (as expected!). I'm using Meltano's sql connector from their [singer-sdk](https://github.com/meltano/sdk/blob/main/singer_sdk/connectors/sql.py). In the `create_sqlalchemy_connection` function they specifically tell users to override that method if the target database doesn't support server side cursors. Thanks again for the help!

**Comment by Mause** [comment]
@Jamiel-Carter this has now been released in v0.6.6 - can you confirm it works as expected? :grinning:
